engine/src/main/java/org/hibernate/search/backend/impl/DeletionQueryMapper.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.backend.impl;
+
+import org.apache.lucene.search.Query;
+import org.hibernate.search.backend.DeletionQuery;
+import org.hibernate.search.util.impl.ScopedAnalyzer;
+
+/**
+ * @author Martin Braun
+ */
+public interface DeletionQueryMapper {
+
+	Query toLuceneQuery(DeletionQuery deletionQuery, ScopedAnalyzer analyzerForEntity);
```

Comment:
I know it's not uncommon and I don't really mind it, but in this case I like the "everything together" approach more.

Also, you also have to look-up the mapper type if you do it with the split approach, but you don't have to lookup the serialization/deserialization code in two different classes.

To be honest, I don't think where this logic is placed shouldn't be that big of a problem for this use case :).